src/server.c

```diff
@@ -405,14 +405,24 @@ int dictExpandAllowed(size_t moreMem, double usedRatio) {
  * In non-cluster mode, we don't need this list as there is only one dictionary per DB. */
 void dictRehashingStarted(dict *d) {
     if (!server.cluster_enabled || !server.activerehashing) return;
+    server.db[0].sub_dict[DB_MAIN].bucket_count += DICTHT_SIZE(d->ht_size_exp[1]); /* Started rehashing (Add the new ht size) */
     listAddNodeTail(server.db[0].sub_dict[DB_MAIN].rehashing, d);
 }
 
+void dictRehashingCompleted(dict *d) {
+    server.db[0].sub_dict[DB_MAIN].bucket_count -= DICTHT_SIZE(d->ht_size_exp[0]); /* Finished rehashing (Remove the old ht size) */
```

Comment:
we're relying on the fact this CB is called before the old table is deleted.
maybe this should be documented in the interface?
maybe in both these callbacks, the dict.h should offer a specific interface to get the rehashing info, like
`void dictGetRehashInfo(long long *from, long long *to);`